### ISSUE TITLE
JWT claim overriding, namespacing or the ability to unreserve claims.

### ISSUE THREAD

**Issue by matthewhartstonge** [FusionAuth/fusionauth-issues#1721]
### Problem
Refer #1669 + https://github.com/FusionAuth/fusionauth-issues/issues/1669#issuecomment-1117011257.

TL;DR: 
- I want to be able to edit all the JWT claims myself. They are mine. ALL MINE. i swearz it.. my precciosusiss..
- I want to remove unneeded JWT payload 'baggage'

### Solution
An ability within FusionAuth to either:
* Remove reservations of certain claims, allowing them to be mutated within a lamba.
* namespace required FusionAuth JWT claims.
* Removal of unneeded/reserved JWT claims via a lambda
* All of the above.

### Alternatives/workarounds

- Read each release note for JWT reservation changes and rearchitect as needed
- Namespace _our_ JWT claims and rearchitect as required
- Stay on the same version of FusionAuth (muhaha #SecOpsTriggered)

### Additional context

We have architected our JWT payloads way back when (circa 2017) to use industry best practices when setting our jwt payloads. For reference, here are the claims we currently use:
- `aud` == Intended audience this JWT was minted for `api.example.com`
- `exp` == token expiry (epoch)
- `iat` == issued at time from (epoch)
- `iss` = Issuer (who gave us this token) `example.fusionauth.io`
- `jti` == JWT (unique) Token UUID
- `nbf` == can't use this token before this time (epoch)
- `scp` == scopes (lol the finalized RFC changed this to `scope` - we're safe!) an array of allowed scopes
- `sub` == user's _person_ UUID
- `tid` == array of allowed tenant access ids.
- `uid` == user's user UUID

Other non-standard claims we happen to use:
- `roles` == array of user granted permissions

Claims our services _base_ validate against (validation+authorization handlers live on top of this based on business rules):
- `exp`
- `iat`
- `iss`
- `nbf`
- `tid`
- `aud`

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by robotdan** [comment]
> I want to be able to edit all the JWT claims myself. They are mine. ALL MINE. i swearz it.. my precciosusiss..

😂

**Comment by robotdan** [comment]
Internal notes:

Possibly add a JWT configuration so we can potentially reserve more things, but the end user has the option to rename our claims so they don't collide with other integrations or they could namespace them.

As an example:

```
Reserved claims:
-----------------------------
Audience: aud
Expiration: exp
Issued At: iat
Roles: roles
Permissions: permissions
Subject: sub
```
